### PR TITLE
Image Registry update for Z and Power

### PR DESCRIPTION
**Pull request by ktania46**
BZ : https://bugzilla.redhat.com/show_bug.cgi?id=1894382

This update is for 4.5 , 4.6 and 4.7 release 

@SNiemann15 @vikram-redhat

### CODE REVIEW

**Comment by ghatwala** [comment]
Had confirmed using this flag - `--binary-image or -i` to prune the source index of all but the specified packages via steps given here - https://bugzilla.redhat.com/show_bug.cgi?id=1894382#c4

**Comment by ghatwala** [comment]
LGTM

**Comment by SNiemann15** [comment]
@doris.xu Can you validate this on Z please?

File: modules/olm-pruning-index-image.adoc [hunk 1]
```diff
@@ -123,10 +123,16 @@ $ opm index prune \
     -f {index-image-pullspec} \// <1>
     -p {package1},{package2},{package3} \// <2>
     -t <target_registry>:<port>/<namespace>/{index-image} <3>
+    -i registry.redhat.io/openshift4/ose-operator-registry/{index-image}
```
Comment (nickboldt) [inline modules/olm-pruning-index-image.adoc:126]: should there be a `<4>` on this line and an explanation below? Also if this is only needed for P & Z, what's the default `-i` value for amd64?

Comment (SNiemann15) [reply modules/olm-pruning-index-image.adoc:126]: If that line isn't needed for other architectures I would suggest we remove the note and make it a callout. 

<4> Use `-i` only with IBM Power Systems and IBM Z images.

Comment (ghatwala) [reply modules/olm-pruning-index-image.adoc:126]: The **default value of base image is** `quay.io/operator-framework/upstream-opm-builder` which isn't multi-arch and thus prune command - `opm index prune ` fails on Power ( should fail on Z too , yet to be validated though) . To resolve this had to replace it with - `registry.redhat.io/openshift4/ose-operator-registry` which is multi-arch passed using flag `-i` to prune command ....  fyi @SNiemann15  @nickboldt

**Comment by ghatwala** [comment]
FYI tdale@ibm.com

File: modules/olm-pruning-index-image.adoc [hunk 2]
```diff
@@ -123,10 +123,13 @@ $ opm index prune \
     -f {index-image-pullspec} \// <1>
     -p {package1},{package2},{package3} \// <2>
     -t <target_registry>:<port>/<namespace>/{index-image} <3>
+    -i registry.redhat.io/openshift4/ose-operator-registry/{index-image} 
 ----
 <1> Index to prune.
 <2> Comma-separated list of packages to keep.
 <3> Custom tag for new index image being built.
+<4> Use `-i` only with IBM Power Systems and IBM Z images.
```
Comment (nickboldt) [inline modules/olm-pruning-index-image.adoc:131]: So then:

<4> Use `-i` only with IBM Power Systems and IBM Z images to specify a multi-arch index image, eg., `-i registry.redhat.io/openshift4/ose-operator-registry/{index-image}`

?

File: modules/olm-pruning-index-image.adoc (hunk 1 above)
Comment (nickboldt) [reply modules/olm-pruning-index-image.adoc:126]: So... leave it in with the default?
```
-i quay.io/operator-framework/upstream-opm-builder <4>
```

then below...

File: modules/olm-pruning-index-image.adoc (hunk 2 above)
Comment (tdaleibm) [reply modules/olm-pruning-index-image.adoc:131]: Will it be mentioned somewhere else what {index-image} to actuallly use? I believe we can explicitly say to use -i registry.redhat.io/openshift4/ose-operator-registry:v4.6 for both P and Z.

[the remark this answers is not in the document]
Comment (tdaleibm) [reply modules/olm-pruning-index-image.adoc:131]: This is required for IBM Z/Power.

**Comment by vikram-redhat** [comment]
hold for merge till after 4.7 GA.

**Comment by codyhoag** [comment]
@ktania46 the edited file in this PR did not exist for `enterprise-4.5`, so the changes were not cherry-picked there. Please follow up with another PR for 4.5 specifically if there's something that needs to be added there. Thanks!

File: modules/olm-pruning-index-image.adoc [hunk 3]
```diff
@@ -123,10 +123,13 @@ $ opm index prune \
     -f {index-image-pullspec} \// <1>
     -p {package1},{package2},{package3} \// <2>
     -t <target_registry>:<port>/<namespace>/{index-image} <3>
+    -i registry.redhat.io/openshift4/ose-operator-registry/{index-image}
```
[the remark this answers is not in the document]
Comment (adellape) [reply modules/olm-pruning-index-image.adoc]: @ktania46 Remember also to put a trailing `\` at the end of previous lines when you're adding a new final line to these sort of multi-line commands. For example:

```
    -t <target_registry>:<port>/<namespace>/{index-image}  \// <3>
    -i registry.redhat.io/openshift4/ose-operator-registry/{index-image} <4>
```

(Note that `\//` is required when the line also includes a callout like `<3>`.)

I'm fixing this spot in another related PR, so just FYI for the future. Thank you for the PR! :tada:

**Comment by jianzhangbjz** [comment]
@codyhoag As you may know, this PR leads to a bug https://bugzilla.redhat.com/show_bug.cgi?id=1943150 reported by the customer. So, could you help ask the QE have a review before merging it in the future? Thanks! cc: @adellape 
You can reach out to us in the `#forum-qe` channel of Slack. Like:
image

**Pull request merged**